contracts/price-providers/V2TokenCollateralPeggedPriceProviderFallback.sol

```diff
@@ -0,0 +1,60 @@
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-eth/contracts/math/SafeMath.sol";
+import "openzeppelin-eth/contracts/math/Math.sol";
+
+import "./PriceProviderFallback.sol";
+import "../interface/IMoCV2.sol";
+import "../interface/IPriceProvider.sol";
+
+/**
+  @notice gets the technical price from V2 Protocol and covert it to pegged price
+  @dev Parameters the MoC V2 address
+  The TP address is need to verified if is the protocol has valid price or not
+*/
+contract V2TokenCollateralPeggedPriceProviderFallback is PriceProviderFallback {
+
+  using SafeMath for uint256;
+  uint256 public constant RATE_PRECISION = uint256(10**18);
+
+  IMoCV2 public mocV2;
+  address public tpAddress;
+
+  constructor(
+    IMoCV2 _mocV2,
+    address _tpAddress,
+    IMoCDecentralizedExchange _dex,
+    address _baseToken,
+    address _secondaryToken
+  ) public PriceProviderFallback(_dex, _baseToken, _secondaryToken) {
+    mocV2 = _mocV2;
+    tpAddress = _tpAddress;
+  }
+
+  function failablePeek() internal view returns (bytes32, bool) {
+
+    IMoCV2.PeggedTokenIndex memory tpIndex = mocV2.peggedTokenIndex(tpAddress);
+
+    if (!tpIndex.exists) {
+      // If not exist is not a valid price
+      return (0, false);
+    }
+
+    IMoCV2.PegContainerItem memory tpItem = mocV2.pegContainer(tpIndex.index);
+
+    IPriceProvider priceProvider = IPriceProvider(tpItem.priceProvider);
+    (bytes32 price, bool isValid) = priceProvider.peek();
```

Comment:
maybe you are using this because is like a template but just in case, remember you can use `mocV2.getPACtp(address tp_)`

Reply:
if we use mocV2.getPACtp(address tp_) what happening in the case of tpAddress not exist anymore  ?